### PR TITLE
Feature/827 context refactor

### PR DESCRIPTION
**Pull request by skuhlmann**
## GitHub Issue

fixes: https://github.com/HausDAO/daohaus-monorepo-deprecated/issues/827

## Changes

- reworks all of moloch-v3-context and it's hooks
- applies across admin app

## Packages Added

n/a

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for _any impacted package_ and check for any errors before the PR)

### CODE REVIEW

File: apps/admin/src/components/Profile.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import {
   DataIndicator,
   widthQuery,
 } from '@daohaus/ui';
-import { AccountProfile } from '@daohaus/utils';
+import { AccountProfile, MolochV3Membership } from '@daohaus/utils';
```
Comment (santteegt) [inline apps/admin/src/components/Profile.tsx:16]: remove unused entity

File: libs/moloch-v3-context/src/utils/fetchHelpers.ts [hunk 2]
```diff
@@ -1,303 +1,206 @@
 import { ReactSetter } from '@daohaus/utils';
```
Comment (santteegt) [inline libs/moloch-v3-context/src/utils/fetchHelpers.ts:1]: remove unused import

**Review by santteegt — APPROVED** [review]

File: apps/admin/src/components/customFields/SelectApplicant.tsx [hunk 3]
```diff
@@ -81,13 +81,13 @@ export const SelectApplicant = ({
...
   useEffect(() => {
     if (members) {
-      // TODO: WHat about pagination?
-      setMemberList(
-        members.map((m) => ({
-          name: m.memberAddress,
-          value: m.memberAddress,
-        }))
-      );
+      // TODO: pagination?
```
Comment (jordanlesich) [inline apps/admin/src/components/customFields/SelectApplicant.tsx:84]: Why the wait on pagination? Does it work?

File: apps/admin/src/pages/Proposals.tsx [hunk 4]
```diff
@@ -102,7 +96,7 @@ export function Proposals() {
   };
 
   useEffect(() => {
-    filterProposals();
+    filterProposals({});
```
Comment (jordanlesich) [inline apps/admin/src/pages/Proposals.tsx:99]: Would it be worthwhile to check first if there's an existing filter before firing filter on page load?

File: libs/moloch-v3-context/src/MolochV3DaoContext.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
+import { Keychain, ValidNetwork } from '@daohaus/keychain-utils';
```
Comment (jordanlesich) [inline libs/moloch-v3-context/src/MolochV3DaoContext.tsx:1]: This context is looking slick. Way easier to follow the thread.

File: libs/moloch-v3-context/src/hooks/useProposals.tsx [hunk 6]
```diff
@@ -1,55 +1,110 @@
...
+    filter: daoData?.proposals?.filter,
     filterProposals,
+    sort: daoData?.proposals?.ordering,
+    sortProposals,
+    paging: {
+      current: daoData?.proposals?.previousPaging,
+      next: daoData?.proposals?.nextPaging,
+    },
+    loadNextPage,
+    refreshProposals,
```
Comment (jordanlesich) [inline libs/moloch-v3-context/src/hooks/useProposals.tsx:108]: I think separating the concerns here was a good idea. This is the busiest one, but still way easier to read.

File: libs/moloch-v3-context/src/hooks/useRecordData.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+// 3. how to make the RecordDataContextType generic to that returned record type is defined when we set up the hook
+
+type RecordDataContextType<T = MolochV3Dao> = {
+  record: T | undefined;
+  refreshEntity: () => Promise<void>;
+};
+
+export const useRecordData = <T, V>(
+  entityName: keyof MolochV3DaoData['dao']
+  // entityName: keyof MolochV3DaoData
```
Comment (jordanlesich) [inline libs/moloch-v3-context/src/hooks/useRecordData.tsx:20]: Remove comments

File: libs/moloch-v3-context/src/utils/fetchHelpers.ts [hunk 8]
```diff
@@ -1,303 +1,206 @@
...
     });
 
-    if (memberRes?.data?.member && shouldUpdate) {
-      setMember(memberRes.data.member);
-    } else if (shouldUpdate) {
-      setMember(undefined);
+    if (res?.data?.member) {
+      return res.data.member;
+    } else {
+      console.error('no member found');
```
Comment (jordanlesich) [inline libs/moloch-v3-context/src/utils/fetchHelpers.ts:66]: I'm wondering if we should add lifecycleFns so we can have some error handling. Could be a ticket for later

**Review by 0xRowdy — APPROVED** [review]

File: apps/admin/src/components/customFields/SelectApplicant.tsx (hunk 3 above)
Comment (skuhlmann) [reply apps/admin/src/components/customFields/SelectApplicant.tsx:84]: > Why the wait on pagination? Does it work?

I just edited the comment. i didn't make any other changes, 

adding pagination to this component is not part of this ticket scope.

File: libs/moloch-v3-context/src/utils/fetchHelpers.ts (hunk 8 above)
Comment (skuhlmann) [reply libs/moloch-v3-context/src/utils/fetchHelpers.ts:66]: Good idea - def makes sense for later

**Pull request merged**